### PR TITLE
[atlona] Fix NPE on shutdown when Thing disabled

### PR DESCRIPTION
**Pull request by mlobstein**
This PR fixes an NPE that occurs when the Thing is disabled and openhab calls dispose() on shutdown. Also fix a large number of warnings.

### CODE REVIEW

File: bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/pro3/AtlonaPro3PortocolHandler.java [hunk 1]
```diff
@@ -1088,7 +1088,7 @@ private class NormalResponseCallback implements SocketSessionListener {
 
         @Override
         public void responseReceived(String response) {
-            if (response == null || response.isEmpty()) {
+            if (response.isEmpty()) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/pro3/AtlonaPro3PortocolHandler.java:1091]: This class is not `@NonNullByDefault`, so I'm wondering if this change introduces risk of NPE? I have not checked usages etc., so just question.

Comment (mlobstein) [reply bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/pro3/AtlonaPro3PortocolHandler.java:1091]: It is only used once in SocketChannelSession and is wrapped with a null check there. I removed the null check here to eliminate the redundant null check warning.
socketchanelsession

Comment (jlaur) [reply bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/pro3/AtlonaPro3PortocolHandler.java:1091]: Much better not to call with null. Thanks for clarifying.

**Review by jlaur — APPROVED** [review]
Thanks for the fix!

**Pull request merged**